src/Framework/Bootloader/Cycle/SchemaBootloader.php

```diff
@@ -107,15 +110,23 @@ public function getGenerators(): array
     protected function schema(MemoryInterface $memory): SchemaInterface
     {
         $schema = $memory->loadData('cycle');
-        if (is_null($schema)) {
+        if (empty($schema)) {
```

Comment:
I think that it can be rewritten as:
```php
$schema = $memory->loadData('cycle');

if ($schema) {
    return new Schema($schema);
}

$schema = (new Compiler())->...
```

Early return will help you get rid of the extra nesting level.

P.S. And the schemа generation can be taken out in a separate private method, then you get result like:
```php
$schema = $memory->loadData('cycle');

return $schema ? new Schema($schema) : $this->createDefaultSchema();
```

Reply:
Refactored the whole operation